File: frontend/src/types/inventaire.ts
export enum CategoryProduit {
  INTRANT = "INTRANT",
  EQUIPEMENT = "EQUIPEMENT",
  RECOLTE = "RECOLTE",
  EMBALLAGE = "EMBALLAGE",
  PIECE_RECHANGE = "PIECE_RECHANGE"
}

export enum UniteMesure {
  KG = "KG",
  LITRE = "LITRE",
  UNITE = "UNITE",
  TONNE = "TONNE",
  METRE = "METRE"
}

export enum TypeMouvement {
  ENTREE = "ENTREE",
  SORTIE = "SORTIE",
  TRANSFERT = "TRANSFERT"
}

export enum PeriodeInventaire {
  JOUR = 'jour',
  SEMAINE = 'semaine',
  MOIS = 'mois',
  ANNEE = 'annee',
  PERSONNALISE = 'personnalise'
}

export interface SeuilStock {
  produit_id: string;
  seuil_min: number;
  seuil_max: number;
  seuil_alerte: number;
  actif: boolean;
  date_modification?: string;
  commentaires?: string;
}

export interface FiltresInventaire {
  periode?: PeriodeInventaire;
  categorie?: CategoryProduit;
  dateDebut?: string;
  dateFin?: string;
  fournisseur?: string;
  type_mouvement?: TypeMouvement;
  recherche?: string;
  skip?: number;
  limit?: number;
}

export interface PrevisionStock {
  date: string;
  valeur_reelle: number;
  valeur_prevue: number;
  categorie?: CategoryProduit;
  produit_id?: string;
}

export interface ConditionsStockage {
  temperature_min: number;
  temperature_max: number;
  humidite_min: number;
  humidite_max: number;
  luminosite_max?: number;
  ventilation_requise: boolean;
}

export interface ConditionsActuelles {
  temperature: number;
  humidite: number;
  luminosite?: number;
  qualite_air?: number;
  derniere_maj: string;
}

export interface Certification {
  nom: string;
  organisme: string;
  date_obtention: string;
  date_expiration: string;
  specifications: Record<string, any>;
}

export interface ControleQualite {
  date_controle: string;
  responsable_id: string;
  resultats: Record<string, any>;
  conforme: boolean;
  actions_requises?: string;
}

export interface Produit {
  id: string;
  code: string;
  nom: string;
  categorie: CategoryProduit;
  description?: string;
  unite_mesure: UniteMesure;
  seuil_alerte?: number;
  prix_unitaire?: number;
  specifications?: Record<string, any>;
  conditions_stockage?: ConditionsStockage;
}

export interface Stock {
  id: string;
  produit_id: string;
  entrepot_id: string;
  quantite: number;
  valeur_unitaire?: number;
  emplacement?: string;
  lot?: string;
  date_peremption?: string;
  origine?: string;
  certifications?: Certification[];
  conditions_actuelles?: ConditionsActuelles;
  capteurs_id?: string[];
  date_derniere_maj: string;
}

export interface MouvementStock {
  id: string;
  produit_id: string;
  type_mouvement: TypeMouvement;
  quantite: number;
  entrepot_source_id?: string;
  entrepot_destination_id?: string;
  responsable_id: string;
  reference_document?: string;
  notes?: string;
  cout_unitaire?: number;
  conditions_transport?: Record<string, any>;
  controle_qualite?: ControleQualite;
  date_mouvement: string;
}

export interface StatsInventaire {
  total_produits: number;
  valeur_totale: number;
  produits_sous_seuil: number;
  mouvements_jour: number;
}

export interface Entrepot {
  id: string;
  nom: string;
  adresse?: string;
  capacite?: number;
  responsable_id?: string;
  conditions_stockage?: ConditionsStockage;
  actif: boolean;
}

export interface InventoryPermissions {
  canCreate: boolean;
  canEdit: boolean;
  canDelete: boolean;
}

File: schemas/inventaire.py
from pydantic import BaseModel, UUID4
from datetime import datetime
from typing import Optional, Dict, List
from enum import Enum

class CategoryProduit(str, Enum):
    INTRANT = "INTRANT"
    EQUIPEMENT = "EQUIPEMENT"
    RECOLTE = "RECOLTE"
    EMBALLAGE = "EMBALLAGE"
    PIECE_RECHANGE = "PIECE_RECHANGE"

class UniteMesure(str, Enum):
    KG = "KG"
    LITRE = "LITRE"
    UNITE = "UNITE"
    TONNE = "TONNE"
    METRE = "METRE"

class TypeMouvement(str, Enum):
    ENTREE = "ENTREE"
    SORTIE = "SORTIE"
    TRANSFERT = "TRANSFERT"

class ConditionsStockage(BaseModel):
    temperature_min: float
    temperature_max: float
    humidite_min: float
    humidite_max: float
    luminosite_max: Optional[float]
    ventilation_requise: bool

class ConditionsActuelles(BaseModel):
    temperature: float
    humidite: float
    luminosite: Optional[float]
    qualite_air: Optional[float]
    derniere_maj: datetime

class Certification(BaseModel):
    nom: str
    organisme: str
    date_obtention: datetime
    date_expiration: datetime
    specifications: Dict

class ControleQualite(BaseModel):
    date_controle: datetime
    responsable_id: UUID4
    resultats: Dict
    conforme: bool
    actions_requises: Optional[str]

class ProduitBase(BaseModel):
    code: str
    nom: str
    categorie: CategoryProduit
    description: Optional[str]
    unite_mesure: UniteMesure
    seuil_alerte: Optional[float]
    prix_unitaire: Optional[float]
    specifications: Optional[Dict]
    conditions_stockage: Optional[ConditionsStockage]

class ProduitCreate(ProduitBase):
    pass

class ProduitResponse(ProduitBase):
    id: UUID4
    
    class Config:
        orm_mode = True

class StockBase(BaseModel):
    produit_id: UUID4
    entrepot_id: UUID4
    quantite: float
    valeur_unitaire: Optional[float]
    emplacement: Optional[str]
    lot: Optional[str]
    date_peremption: Optional[datetime]
    origine: Optional[str]
    certifications: Optional[List[Certification]]
    conditions_actuelles: Optional[ConditionsActuelles]
    capteurs_id: Optional[List[UUID4]]

class StockResponse(StockBase):
    id: UUID4
    date_derniere_maj: datetime
    
    class Config:
        orm_mode = True

class MouvementStockBase(BaseModel):
    produit_id: UUID4
    type_mouvement: TypeMouvement
    quantite: float
    entrepot_source_id: Optional[UUID4]
    entrepot_destination_id: Optional[UUID4]
    responsable_id: UUID4
    reference_document: Optional[str]
    notes: Optional[str]
    cout_unitaire: Optional[float]
    conditions_transport: Optional[Dict]
    controle_qualite: Optional[ControleQualite]

class MouvementStockCreate(MouvementStockBase):
    pass

class MouvementStockResponse(MouvementStockBase):
    id: UUID4
    date_mouvement: datetime
    
    class Config:
        orm_mode = True
